src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1084,7 +1086,7 @@ def _update_container_app_source(self, cmd, source, build_env_vars, registry_ser
         app.run_source_to_cloud_flow(source, dockerfile, build_env_vars, can_create_acr_if_needed=False, registry_server=registry_server)
 
         # Validate an image associated with the container app exists
-        containers = safe_get(self.containerapp_def, "properties", "template", "containers", default=[])
+        containers = safe_get(self.new_containerapp, "properties", "template", "containers", default=[])
```

Comment:
Could you please clarify where is this new_containerapp defined? Do we even need this with your other change above?

Reply:
This change is not related to the bugfix in this PR but I think it makes sense to change this to `self.new_containerapp` instead because this payload was already populated before it reached this flow (since this is the preview update flow) so keeping it as `self.containerapp_def` might not have the changes from `self.new_containerapp` in some cases.

This is not a regression so I can remove this and create a separate PR if it makes more sense.